---
title: "MSU - Week 3: Statistical models of Differential Expression"
date: "08/15/2015"
output: html_document
layout: topic
minutes: 20
---

# The mean-variance relationship (and what it means)

Let's start by loading the data:
```{r load}
load('bottomly_eset.RData')
```

Take a look at your environment, and you should find that three ExpressionSet objects have been loaded along with all the required libraries. _If you have errors with this, you may have to re-install the package. Please take a look at the [README](./README.md) for more details on requirements for this module._

The three datasets correspond to the full [Bottomly](http://www.ncbi.nlm.nih.gov/pubmed?term=21455293) dataset that is available on [ReCount](http://bowtie-bio.sourceforge.net/recount/), and two smaller datasets generated by randomly selecting samples from the full set. The full set contains data from two different mouse strains C57Bl and DBA, with 10 and 11 replicates, respectively. We created another dataset with 5 replicates for each strain (n = 10), and one with 2 replicates for each strain (n = 4). **How does the relationship between mean and variance change as you increase the number of replicates?**

```{r mean-variance, fig.align='center'}

# Plot 2 replicate dataset
eset <- bottomly.2reps
cpm.mat <- log(cpm(exprs(eset)))
mean.vec <- apply(cpm.mat, 1, mean)
sdvec <- apply(cpm.mat, 1, sd)
plot(mean.vec, sdvec, pch=".", main="2 replicates", ylab="sd", xlab="Average logCPM")

# Plot 5 replicate dataset
eset <- bottomly.5reps
cpm.mat <- log(cpm(exprs(eset)))
mean.vec <- apply(cpm.mat, 1, mean)
sdvec <- apply(cpm.mat, 1, sd)
plot(mean.vec, sdvec, pch=".", main="5 replicates", ylab="sd", xlab="Average logCPM")

# Plot 10 replicate dataset
eset <- bottomly.eset
cpm.mat <- log(cpm(exprs(eset)))
mean.vec <- apply(cpm.mat, 1, mean)
sdvec <- apply(cpm.mat, 1, sd)
plot(mean.vec, sdvec, pch=".", main="10 replicates", ylab="sd", xlab="Average logCPM")

```

Appropriate modeling of the mean-variance relationship in DGE data is important for making inferences about differential expressionIn RNA-Seq data, genes with larger average expression have on average larger observed variances across samples, that is, they vary in expression from sample to sample more than other genes with lower average expression. This phenomena of 'having different scatter' is known as data **heteroscedasticity**. In our figures, we observe this heteroscedasticity except that the logarithmic transformation counteracts this and overdoes the adjustment somewhat so that large log-counts now have smaller scatter than small log-counts. More, importantly we see the scatter tends to reduce as we increase the number of replicates.



